### PR TITLE
Hashed transaction duplicates

### PR DESCRIPTION
**Pull request by maxwofford**

### CODE REVIEW

**Comment by maxwofford** [comment]
Closes #976, #978, #975

**Review by motdotla — DISMISSED** [review]

**Comment by maxwofford** [comment]
Also includes an initial version of #980

File: app/services/transaction_engine/canonical_transaction_service/import/plaid_that_look_like_duplicates.rb [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+    module Import
+      class PlaidThatLookLikeDuplicates
+        def run
+          unprocessed_plaid_with_duplicate_hashes.find_each do |ht|
+            # For some reason we imported transactions from the same bank
+            # account through 2 different plaid connections. All those
+            # transactions would be actual duplicates (they'd each have 1
+            # transaction from each plaid connection)
+
+            next unless true
```
Comment (motdotla) [inline app/services/transaction_engine/canonical_transaction_service/import/plaid_that_look_like_duplicates.rb]: is this necessary?

File: app/services/transaction_engine/hashed_transaction_service/grouped_duplicates.rb [hunk 2]
```diff
@@ -0,0 +1,9 @@
+module TransactionEngine
+  module HashedTransactionService
+    class GroupedDuplicates
+      def run
+        ::TransactionEngine::HashedTransactionService::Duplicates.new.run.group_by(&:primary_hash).values
```
Comment (motdotla) [inline app/services/transaction_engine/hashed_transaction_service/grouped_duplicates.rb]: :+1:

**Review by motdotla — DISMISSED** [review]
looks good. just one nitpick

File: app/services/transaction_engine/canonical_transaction_service/import/plaid_that_look_like_duplicates.rb (hunk 1 above)
Comment (maxwofford) [reply app/services/transaction_engine/canonical_transaction_service/import/plaid_that_look_like_duplicates.rb]: Nope, accidentally added while debugging. Removed

**Comment by maxwofford** [comment]
@motdotla Ok, this is ready for review. I didn't like the way `not_actual_duplicate` works because I'm pretty sure we're missing a case or two, but I'm fine to merge what's currently here if it means we can unblock @lachlanjc. Any edge case will get surfaced once we have the front-end for resolving duplicates

**Comment by maxwofford** [comment]
Merging after green light from Scott on Slack

**Pull request merged**